packages/vscode-extension/src/webview/views/CreateDeviceView.tsx

```diff
@@ -100,39 +91,45 @@ function CreateDeviceView({ onCreate, onCancel }: CreateDeviceViewProps) {
         if (!runtime) {
           return;
         }
-        const iOSDeviceType = runtime.supportedDeviceTypes.find(({ name }) => name === displayName);
+        const iOSDeviceType = runtime.supportedDeviceTypes.find(
+          ({ identifier }) => identifier === deviceModelId
+        );
         if (!iOSDeviceType) {
           return;
         }
-        await deviceManager.createIOSDevice(iOSDeviceType, runtime);
+        await deviceManager.createIOSDevice(iOSDeviceType, displayName.trim(), runtime);
       } else {
         const systemImage = androidImages.find((image) => image.location === selectedSystemName);
-        if (!systemImage || !displayName || !deviceName) {
+        if (!systemImage) {
           return;
         }
-        await deviceManager.createAndroidDevice(displayName, deviceName, systemImage);
+        await deviceManager.createAndroidDevice(deviceModelId, displayName.trim(), systemImage);
       }
     } finally {
       onCreate();
     }
   }
 
+  const handleDisplayNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const text = formatDisplayName(event.target.value);
+    setDisplayName(text);
+    setIsDisplayNameValid(0 < text.length && text.length <= MAX_DEVICE_NAME_LENGTH);
+  };
+
   return (
     <div className="edit-device-form">
       <div className="form-row">
         <Label>Device Type</Label>
         <Select
           className="form-field"
-          value={`${
-            devicePlatform && displayName ? `${devicePlatform}:${displayName}:${deviceName}` : ""
-          }`}
-          onChange={(newValue: string) => {
-            const [newPlatform, newDisplayName, newDeviceName] = newValue.split(":", 3);
-            assertPlatform(newPlatform);
-            setDisplayName(newDisplayName);
-            setDeviceName(newDeviceName);
-            setDevicePlatform(newPlatform);
+          value={deviceModelId ?? ""}
+          onChange={(modelId: string) => {
+            setDevicePlatform(modelId.startsWith("com.apple") ? "ios" : "android");
+            setDeviceModelName(mapIdToModel(modelId));
+            setDeviceModelId(modelId);
```

Comment:
seems like it'd be much simpler to just keep DeviceProperties object in the state. Then we can have a method that finds it based on the model id and updates the state. We can extract devicePlatform, modelId and modelName from it